### ISSUE TITLE
benchmark: Wang dataset

### ISSUE THREAD

**Issue by jacenkow** [jacenkow/beard-server#3]
```beard-server``` RESTful API was measured against time, with the model retrained on [Wang dataset](https://github.com/inveniosoftware/beard/tree/master/examples/data) (9479 records in total).

The machine used to for the testing has following specifications:

inspire-ml-01 | 
------------ |
16x Intel Xeon E312xx (Sandy Bridge) |
32GB RAM |
Python (2.7.5) |
numpy (1.10.1) |
scikit-learn (0.16.1) |
scipy (0.16.1) |

The time measurement was as following for the given features during building a distance estimator:

| Features | |
| ------------ | ------------ |
| author_full_name | author_second_initial_similarity |
| author_first_given_name_similarity | author_second_given_name_similarity |
| author_other_names_similarity | affiliation_similarity |
| coauthors_similarity | title_similarity |
| journal_similarity | abstract_similarity |
| keywords_similarity | collaborations_similarity |
| subject_similairty | year_diff |

| ```time python curl_training.py``` | ```time python curl_clustering.py``` |
| ------------- | -------------: |
| real    104m11.530s | real    16m14.869s |
| user    0m2.907s | user    0m3.138s |
| sys     0m0.516s | sys     0m0.635s |

**Comment by jacenkow** [comment]
![screen shot 2015-11-09 at 17 50 24](https://cloud.githubusercontent.com/assets/12476583/11039963/664c12cc-870a-11e5-9eea-ced4322b973f.png)

**Comment by kaplun** [comment]
fast=2 is really promising. Would be interesting to try one day, to recompute these numbers excluding collaboration papers. I have the suspect that co-authors are highly impacted by the exisiting of collaboration papers. Additionally would be interesting to see how much adding year back to fast=2 is impacting on the performance. It could help a lot to avoid false positives. (and also to compensate when 2 have different affiliation, but also very different year, hence they could be the same person).

**Comment by jacenkow** [comment]
@kaplun testing ```year_diff``` now :+1:

**Comment by jacenkow** [comment]
| Features | ```fast = 2``` with ```year_diff``` |
| ------------ | ------------ |
| author_full_name | affiliation_similarity |
| coauthors_similarity | year_diff |

| ```time python curl_training.py``` | ```time python curl_clustering.py``` |
| ------------- | -------------: |
| real   103m33.408s | |
| user    0m3.010s | |
| sys    0m0.563s | |

**Comment by kaplun** [comment]
Uh: it took longer time to do the training.

**Comment by jacenkow** [comment]
Yeah, it is super weird for me, that's why I will **not** use ```screen``` (funny stuff about the system shows up) this time and just do an ugly pipe and retrain again. :smile:

**Comment by jacenkow** [comment]
@kaplun seems we should stick to ```fast = 2``` without ```year_diff``` feature.

**Comment by kaplun** [comment]
wow. the year is so expensive?

**Comment by jacenkow** [comment]
Je ne sais pas

**Comment by glouppe** [comment]
Just a few comments to keep in mind when doing this kind of study:
- Time of training is irrelevant. This is something you have to do once in a while, hence going from 3 hours to 1 hour has very limited value. 
- You should update to scikit-learn 0.17, as we improve quite a few things to make training faster. 
- When comparing different feature sets, please also evaluate the goodness of the clustering on held-out data. I would put more emphasis on being correct than on being fast (and wrong). 
- The wang data may not be a good representative of the whole dataset, both regarding accuracy and time. Therefore I would be cautious not to draw any general conclusion from this. I would recommend instead following a protocol similar to what we did in the paper.

**Comment by jacenkow** [comment]
@glouppe thank you, really appreciate it. :smile:
